src/main/java/ai/spice/SpiceClient.java

```diff
@@ -185,6 +193,56 @@ public void refresh(String dataset) throws ExecutionException {
         }
     }
 
+    /**
+     * Refreshes an accelerated dataset using the configured dataset acceleration
+     * settings
+     * 
+     * @param dataset        the name of the dataset to refresh
+     * @param refreshOptions the refresh options to use when refreshing the dataset
+     * @throws ExecutionException if there is an error refreshing the dataset
+     */
+    public void refresh_dataset(String dataset, RefreshOptions refreshOptions) throws ExecutionException {
```

Comment:
Most of the functionality duplicates `refresh_dataset(String dataset)` - can we re-use functionality or just have single method with `RefreshOptions refreshOptions` optional parameter / can be null

Reply:
`refresh_dataset(String dataset)` can just call `refresh_dataset(String dataset, RefreshOptions refreshOptions)` and pass null second option